workspaces/copilot/plugins/copilot-backend/src/processors/OrganizationTeamProcessor.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MetricsType } from '@backstage-community/plugin-copilot-common';
+import { DatabaseHandler, MetricDbRow } from '../db/DatabaseHandler';
+import { IProcessor } from './IProcessor';
+import { GithubClient } from '../client/GithubClient';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import {
+  filterNewMetrics,
+  prepareMetricsForInsert,
+} from '../utils/metricHelpers';
+import { batchInsertInChunks } from '../utils/batchInsert';
+import { Config } from '@backstage/config';
+
+type Options = {
+  api: GithubClient;
+  logger: LoggerService;
+  db: DatabaseHandler;
+};
+
+export class OrganizationTeamProcessor implements IProcessor {
```

Comment:
I've noticed that the logic for all 4 classes here is very similar. Perhaps we could transform them in reusable functions? For example `OrganizationTeamProcessor` can become -> `discoverTeamsMetrics` and `OrganizationProcessor` -> `discoverOrganizationMetrics`:

```ts
function discoverTeamsMetrics({ api }) {
  const teams = await api.fetchOrganizationTeams();
  for (const team of teams) {
    discoverOrganizationMetrics(team);
  }
}
```

The naming also confuses me, as "processor" is a concept around catalog, while these are more "tasks" and have nothing to do with catalog. The `IProcessor` interface can be removed as well as it doesn't add so much value: `name` is never used and `shouldRun()` it's just a check that can be moved inside the function.